vertx-web/src/main/java/io/vertx/ext/web/sstore/impl/SessionImpl.java

```diff
@@ -74,6 +75,18 @@ public String id() {
   }
 
   @Override
+  public Session regenerateId() {
+    if (oldId == null) {
+      // keep track of just the first one since the
+      // regeneration during the remaining lifecycle are ephemeral
+      oldId = id;
+    }
+    id = UUID.randomUUID().toString();
```

Comment:
@pmlopes even on a budget we can avoid some of the problems. There is a SHA1PRNG algorithm which, if seeded in advance, is non-blocking. Rather than using `new SecureRandom()` I often declare the dependency in the constructor and have some higher level code provide it from a method such as the one below. That allows better control over the policies without having to reach too deeply into many areas of the code.

```java
  /**
   * Creates and seeds a secure random number generator, and registers a
   * timer in vertx to re-seed it periodically.
   *
   * <p>DO NOT call this method repeatedly. Call it once, and pass the
   * object around.</p>
   *
   * @param vertx a periodic timer will be registered with this vertx instance
   */
  @Nonnull
  public static SecureRandom createSecureRandom(@Nonnull Vertx vertx) {
    // The SHA-1 implementation is seeded via Native (/dev/random) but provides
    // non-blocking output with good cryptographic properties
    SecureRandom newRandom;
    try {
      newRandom = SecureRandom.getInstance("SHA1PRNG");
    } catch (NoSuchAlgorithmException e) {
      newRandom = new SecureRandom();
      log.warn("Unable to use SHA1PRNG for secure random source, defaulted to: " + newRandom.getAlgorithm() + " "
          + newRandom.getProvider(), e);
    }
    final SecureRandom random = newRandom;

    // Make sure default seeding happens now to avoid calling setSeed() too early
    random.nextBoolean();

    // Add a little more seed randomness every five minutes
    vertx.setPeriodic(300000L, id -> executeBlocking(vertx, f -> {
      Metric metric = new Metric(log.isTraceEnabled());
      random.setSeed(random.generateSeed(4));
      if (log.isTraceEnabled()) {
        log.trace("Re-seeded secure random " + metric.getMessage());
      }
      f.complete();
    }, r -> {
      if (r.failed()) {
        log.warn("Problem re-seeding secure random", r.cause());
      }
    }));

    return random;
  }
```